Add proposals dashboard, filter modal

0xLienid opened the pull request.

0xLienid committed:
Add proposals dashboard, frame of filter modal

appleseed-iii committed:
Merge branch 'governance' into ocg/proposals-dashboard

appleseed-iii commented:
merged in develop->governance->`this branch` via https://github.com/OlympusDAO/olympus-frontend/pull/2041/commits/e651e05b5b5c929979889b2af633381e984deb42 because develop add some annoying wagmi/rainbowkit updates that would cause us more merge conflicts later

appleseed-iii committed:
remove childPaperBackground compile issue

appleseed-iii committed:
mocked up modal

appleseed-iii committed:
wip - filtration work in progress

0xLienid committed:
Set up proposal page routing and frame proposal page component

0xLienid committed:
Modified alignment

0xLienid committed:
Styling and tab functionality

0xLienid committed:
Placeholder end date countdown

0xLienid committed:
Switch from x to left chevron for back button

0xLienid committed:
Adding content for votes section

0xLienid committed:
Start create proposal form

0xLienid committed:
Clean up create proposals page code

0xLienid committed:
Add continue button

0xLienid committed:
Clean up proposals page implementation

appleseed-iii committed:
Merge pull request #2083 from OlympusDAO/ocg/proposals-page

Proposals Page and Create Proposal

0xLienid committed:
Mock proposal states

0xLienid committed:
Update snapshots, unsure why Give was failing

appleseed-iii committed:
Merge pull request #2087 from OlympusDAO/ocg/proposal-state

Mock proposal states

appleseed-iii committed:
proposals state

appleseed-iii committed:
Merge branch 'ocg/proposals-dashboard' of github-appleseed:OlympusDAO/olympus-frontend into ocg/proposals-dashboard

brightiron committed:
add contract abis

brightiron committed:
add contract addresses

brightiron committed:
query hooks

brightiron committed:
annotations. add query hooks for contracts

brightiron committed:
OCG Sync Alpha  (#2113)

* updating alpha snapshots

* build(deps-dev): bump @babel/plugin-proposal-private-methods (#2038)

Bumps [@babel/plugin-proposal-private-methods](https://github.com/babel/babel/tree/HEAD/packages/babel-plugin-proposal-private-methods) from 7.17.12 to 7.18.6.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.18.6/packages/babel-plugin-proposal-private-methods)

---
updated-dependencies:
- dependency-name: "@babel/plugin-proposal-private-methods"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* updating v3 colors

* zap input styling to align with other inputs

* build(deps-dev): bump @babel/preset-react from 7.17.12 to 7.18.6 (#2052)

Bumps [@babel/preset-react](https://github.com/babel/babel/tree/HEAD/packages/babel-preset-react) from 7.17.12 to 7.18.6.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.18.6/packages/babel-preset-react)

---
updated-dependencies:
- dependency-name: "@babel/preset-react"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* updating component library

* build(deps-dev): bump eslint-plugin-prettier from 4.0.0 to 4.2.1 (#2057)

Bumps [eslint-plugin-prettier](https://github.com/prettier/eslint-plugin-prettier) from 4.0.0 to 4.2.1.
- [Release notes](https://github.com/prettier/eslint-plugin-prettier/releases)
- [Changelog](https://github.com/prettier/eslint-plugin-prettier/blob/master/CHANGELOG.md)
- [Commits](https://github.com/prettier/eslint-plugin-prettier/compare/v4.0.0...v4.2.1)

---
updated-dependencies:
- dependency-name: eslint-plugin-prettier
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* remove apy copy & replace with annual rebases (#2054)

* remove apy copy & replace with annual rebases

* update next rebase amount to your next rebase

* adjusting tooltip APY reference

Co-authored-by: brightiron <brightiron@protonmail.com>

* build(deps-dev): bump @babel/runtime from 7.18.3 to 7.18.6 (#2058)

Bumps [@babel/runtime](https://github.com/babel/babel/tree/HEAD/packages/babel-runtime) from 7.18.3 to 7.18.6.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.18.6/packages/babel-runtime)

---
updated-dependencies:
- dependency-name: "@babel/runtime"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* build(deps-dev): bump @babel/plugin-transform-destructuring (#2043)

Bumps [@babel/plugin-transform-destructuring](https://github.com/babel/babel/tree/HEAD/packages/babel-plugin-transform-destructuring) from 7.18.0 to 7.18.6.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.18.6/packages/babel-plugin-transform-destructuring)

---
updated-dependencies:
- dependency-name: "@babel/plugin-transform-destructuring"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* build(deps-dev): bump @babel/plugin-proposal-class-properties (#2037)

Bumps [@babel/plugin-proposal-class-properties](https://github.com/babel/babel/tree/HEAD/packages/babel-plugin-proposal-class-properties) from 7.17.12 to 7.18.6.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.18.6/packages/babel-plugin-proposal-class-properties)

---
updated-dependencies:
- dependency-name: "@babel/plugin-proposal-class-properties"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* updating dark color palette. changing background and nav

* updating snapshots

* update snapshots

* add test back

* update test

* build(deps-dev): bump @babel/plugin-syntax-flow from 7.17.12 to 7.18.6 (#2063)

Bumps [@babel/plugin-syntax-flow](https://github.com/babel/babel/tree/HEAD/packages/babel-plugin-syntax-flow) from 7.17.12 to 7.18.6.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.18.6/packages/babel-plugin-syntax-flow)

---
updated-dependencies:
- dependency-name: "@babel/plugin-syntax-flow"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* build(deps-dev): bump @typescript-eslint/eslint-plugin (#2062)

Bumps [@typescript-eslint/eslint-plugin](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/eslint-plugin) from 5.30.0 to 5.30.6.
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v5.30.6/packages/eslint-plugin)

---
updated-dependencies:
- dependency-name: "@typescript-eslint/eslint-plugin"
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* build(deps-dev): bump @lingui/macro from 3.13.3 to 3.14.0 (#2044)

Bumps [@lingui/macro](https://github.com/lingui/js-lingui) from 3.13.3 to 3.14.0.
- [Release notes](https://github.com/lingui/js-lingui/releases)
- [Changelog](https://github.com/lingui/js-lingui/blob/main/CHANGELOG.md)
- [Commits](https://github.com/lingui/js-lingui/compare/v3.13.3...v3.14.0)

---
updated-dependencies:
- dependency-name: "@lingui/macro"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* build(deps): bump @mui/material from 5.8.3 to 5.8.7 (#2040)

* build(deps): bump @mui/material from 5.8.3 to 5.8.7

Bumps [@mui/material](https://github.com/mui/material-ui/tree/HEAD/packages/mui-material) from 5.8.3 to 5.8.7.
- [Release notes](https://github.com/mui/material-ui/releases)
- [Changelog](https://github.com/mui/material-ui/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mui/material-ui/commits/v5.8.7/packages/mui-material)

---
updated-dependencies:
- dependency-name: "@mui/material"
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

* update snapshots

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: brightiron <brightiron@protonmail.com>

* build(deps): bump wagmi from 0.5.6 to 0.5.9 (#2074)

Bumps [wagmi](https://github.com/tmm/wagmi) from 0.5.6 to 0.5.9.
- [Release notes](https://github.com/tmm/wagmi/releases)
- [Commits](https://github.com/tmm/wagmi/compare/wagmi@0.5.6...wagmi@0.5.9)

---
updated-dependencies:
- dependency-name: wagmi
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* build(deps-dev): bump jest-puppeteer from 6.1.0 to 6.1.1 (#2072)

Bumps [jest-puppeteer](https://github.com/smooth-code/jest-puppeteer) from 6.1.0 to 6.1.1.
- [Release notes](https://github.com/smooth-code/jest-puppeteer/releases)
- [Changelog](https://github.com/smooth-code/jest-puppeteer/blob/master/CHANGELOG.md)
- [Commits](https://github.com/smooth-code/jest-puppeteer/compare/v6.1.0...v6.1.1)

---
updated-dependencies:
- dependency-name: jest-puppeteer
  dependency-type: direct:development
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* build(deps-dev): bump @babel/plugin-proposal-private-property-in-object (#2075)

Bumps [@babel/plugin-proposal-private-property-in-object](https://github.com/babel/babel/tree/HEAD/packages/babel-plugin-proposal-private-property-in-object) from 7.17.12 to 7.18.6.
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.18.6/packages/babel-plugin-proposal-private-property-in-object)

---
updated-dependencies:
- dependency-name: "@babel/plugin-proposal-private-property-in-object"
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* build(deps-dev): bump eslint from 8.18.0 to 8.19.0 (#2071)

Bumps [eslint](https://github.com/eslint/eslint) from 8.18.0 to 8.19.0.
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v8.18.0...v8.19.0)

---
updated-dependencies:
- dependency-name: eslint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

* update snapshot

Co-authored-by: appleseed-iii <80423742+appleseed-iii@users.noreply.github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

appleseed-iii committed:
add Governance Instructions contract

appleseed-iii committed:
wip modifying hooks to query proposals

appleseed-iii committed:
update contract parsing for proposals, need metadata next

appleseed-iii committed:
fix loading

appleseed-iii committed:
proposal content.metadata parsing

appleseed-iii committed:
markdown parsing

brightiron committed:
fix image overflow issue

brightiron committed:
remove unused code

brightiron committed:
temp add back as other components are using it

brightiron committed:
a tags should use mui link component to receive styling

appleseed-iii committed:
web3Storage functionality, need to tie to form

appleseed-iii committed:
Merge branch 'ocg/proposals-dashboard' into ocg/contracts

appleseed-iii committed:
wip - connected ipfs to form, needs cleanup

brightiron committed:
updating creation form

brightiron committed:
Merge branch 'ocg/contracts' of https://github.com/OlympusDAO/olympus-frontend into ocg/contracts

brightiron committed:
refactor markdown preview to reusable component

brightiron committed:
allowedImgDomains list. Hides proposal images by default that arent on approved domain lists

brightiron committed:
refactor / update proposal view based on latest design

brightiron committed:
full width for vote page

brightiron committed:
dashboard

brightiron committed:
link

brightiron committed:
votes tab

brightiron committed:
updating component library

appleseed-iii committed:
ugly, but functioning contract creation

appleseed-iii committed:
Merge branch 'ocg/contracts' of github-appleseed:OlympusDAO/olympus-frontend into ocg/contracts

brightiron committed:
rename routes to /governance. add sidenav

brightiron committed:
moving from tabs to routes

brightiron committed:
Merge branch 'ocg/contracts' of https://github.com/OlympusDAO/olympus-frontend into ocg/contracts

brightiron committed:
Merge pull request #2114 from OlympusDAO/ocg/contracts

OCG Contract Integrations

brightiron committed:
Merge branch 'alpha' into ocg/proposals-dashboard

0xLienid committed:
Initial commit for voting hooks

0xLienid committed:
Update voting hooks, tie in vote hook to VotesTab

0xLienid committed:
Bump useProposal and useProposals to react query 4

0xLienid committed:
Merge pull request #2140 from OlympusDAO/ocg/react-query-fix

Bump useProposal and useProposals to react query 4

0xLienid committed:
Merge branch 'ocg/proposals-dashboard' into ocg/voting

0xLienid committed:
Fix VotesTab onClick functions

0xLienid committed:
Add button fill on vote selection

brightiron committed:
use wallet balance. disable vote buttons if not active. or if not connected

brightiron committed:
resolves unable to delete all content from markdown editor

brightiron committed:
must have a value error

brightiron committed:
console log

brightiron committed:
adding rehype sanitize

brightiron committed:
Merge branch 'ocg/proposals-dashboard' into ocg/voting

brightiron committed:
Merge pull request #2148 from OlympusDAO/ocg/voting

Add hooks for voting and endorsing

brightiron committed:
delegate modal. refactor to avoid duplication

0xLienid committed:
Add verification of valid address to delegate modal

0xLienid committed:
Fix misalignment of ProposalSkeleton

brightiron committed:
Merge pull request #2154 from OlympusDAO/ocg/skeleton-fix

Fix misalignment of ProposalSkeleton

0xLienid committed:
switch to isValidAddress helper

appleseed-iii committed:
Merge pull request #2150 from OlympusDAO/ocg/delegateModal

delegate modal. refactor to avoid duplication

brightiron committed:
Merge pull request #2153 from OlympusDAO/ocg/delegateModal-address-entry

Add verification of valid address to delegate modal

0xLienid committed:
Fix alignment on ProposalContainer

0xLienid committed:
Merge pull request #2177 from OlympusDAO/ocg/alignment-fix

Fix alignment on ProposalContainer

0xLienid committed:
Fix bug noted on proposal time remaining

0xLienid committed:
Tweak display of time remaining on proposal page

0xLienid committed:
Copy tweak

appleseed-iii committed:
Merge pull request #2178 from OlympusDAO/ocg/proposal-time-remaining-fix

End time and time remaining fixes

appleseed-iii committed:
replace relative paths w absolute

appleseed-iii committed:
update contract addresses & abi

appleseed-iii committed:
more abi & type updates / tuneup

appleseed-iii committed:
update proposal parsing

appleseed-iii committed:
update uniswap test for random tvl error

appleseed-iii committed:
Merge branch 'governance' into ocg/proposals-dashboard

appleseed-iii commented:
merging this into governance & will create prs off of that
